bin/run.dart

```diff
@@ -19,12 +19,19 @@ const Map<String, String> iconFileNames = const <String, String>{
 
 LinkProvider link;
 
+bool secureMode = false;
 bool enableDsaDiagnosticMode = false;
 
 typedef Action(Map<String, dynamic> params);
 typedef ActionWithPath(Path path, Map<String, dynamic> params);
 
-addAction(handler) {
+addAction(handler, [bool isInsecure = false]) {
```

Comment:
If you return a function here, you can add it the the signature. Maybe even use typedefs if it's clearer.